x/cheqd/keeper/msg_server_did.go

```diff
@@ -13,7 +13,7 @@ import (
 
 func (k msgServer) CreateDid(goCtx context.Context, msg *types.MsgCreateDid) (*types.MsgCreateDidResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
-	prefix := types.DidPrefix + ":" + types.DidMethod + ":" + ctx.ChainID() + ":"
+	prefix := types.DidPrefix + ":" + types.DidMethod + ":" + k.GetDidNamespace(ctx) + ":"
```

Comment:
And if the namespace is empty, handle it gracefully. I mean not put the second `:` in a row.